provider/testdata/sources.toml

```diff
@@ -0,0 +1,252 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+# icu4x_testdata metadata: schema defined in ./src/metadata.rs
+
+# Locales to include in testdata. Keep this list somewhat short, but cover all features.
+# We use 10 base languages with a small number of variants to allow for 10 languages to be
+# used in microbenchmarks.
+locales = [
+    # Arabic:
+    # - Good example for RTL
+    # - Non-latin numerals in Egypt
+    "ar",
+    "ar-EG",
+
+    # Bangla:
+    # - Uses non-Latin numerals
+    "bn",
+
+    # Chakma:
+    # - High-coverage language that uses non-BMP code points
+    "ccp",
+
+    # English:
+    # - Widely understood language in software engineering
+    # - Includes regional variants to test similar-data fallbacks
+    "en",
+    "en-001",
+    "en-ZA",
+
+    # Spanish:
+    #  - Most popular Romance language
+    #  - South American dialect
+    #  - Has context dependent list fragments
+    "es",
+    "es-AR",
+
+    # French:
+    # - Often the first non-English locale to receive new data in CLDR
+    "fr",
+
+    # Filipino:
+    # - Week of month/year have plural variants.
+    "fil",
+
+    # Japanese:
+    # - Four scripts
+    # - Complex date patterns
+    "ja",
+
+    # Russian:
+    # - Cyrillic script
+    # - Interesting plural rules
+    # - Hightly inflected, many gramatical cases
+    "ru",
+
+    # Serbian:
+    # - Multiple scripts
+    # - Southern Europe
+    # - Hightly inflected, many gramatical cases
+    "sr",
+    "sr-Cyrl",
+    "sr-Latn",
+
+    # Thai:
+    # - Complex word breaking
+    "th",
+
+    # Turkish:
+    # - Interesting case-mappings
+    "tr",
+
+    # Root data
+    "und",
+]
+
+# Paths from CLDR JSON to copy into testdata. Uses gitignore-like syntax.
+# The variable "$LOCALES" is replaced with the list of locales from above.
+cldr_json_glob = [
```

Comment:
Do we need to link in the globs? They're only used by the download binary, which is more of a script for us than an actual publishable binary.

In this file I think they distract from the core information: locales and versioning. The globs don't really provide any functionality, they're more of an optimisation to reduce crate size. Maybe we can move them into a separate file that's loaded by the binary?

Reply:
Hmm. An argument to keep them in here is that we should be able to hash this file as a key to determine whether testdata-download-sources needs to be re-run (#2880). That depends on the globs.

Reply:
We need to monitor at least `download-sources.rs` as well, so we could move them in there or have a third file.

Reply:
Not sure why we need to monitor `download-sources.rs`?

Reply:
Changes to the download logic should trigger the CI as well.

Reply:
I'll merge the PR since it blocks 1.1 and we don't yet have consensus on the path forward. We can easily change this in the future.